xamarin/Xamarin.Forms
# Fix additional scenarios with nested defined routes

### Description of Change ###

When defining routes like so 

```C#
			Routing.RegisterRoute("page1/page2", typeof(ShellTestPage));
			Routing.RegisterRoute("page1/page2/page3", typeof(TestPage1));
			var shell = new TestShell(
				CreateShellItem(shellSectionRoute: "page1")
			);
```

Shell wasn't correctly collapsing page1 from the shellsectionroute into the global route because of the default route set on the Content

### Issues Resolved ### 
- fixes #13611 


### Platforms Affected ### 
- Core/XAML (all platforms)


### Testing Procedure ###
- Unit tests included to match reported issue
- download the repro from the referenced issue and make sure nugets work as expected

### PR Checklist ###
